src/master-playlist-controller.js

```diff
@@ -651,7 +653,12 @@ export class MasterPlaylistController extends videojs.EventTarget {
       this.tryToCreateSourceBuffers_();
     } catch (e) {
       videojs.log.warn('Failed to create Source Buffers', e);
-      return this.mediaSource.endOfStream('decode');
+      if (this.mediaSource.readyState !== 'open') {
+        this.trigger('error');
+      } else {
+        this.sourceUpdater_.endOfStream('decode');
```

Comment:
Previously an error would not be shown if endOfStream failed. `endOfStream` should no longer fail.